mindmeld/components/entity_resolver.py

```diff
@@ -562,38 +776,829 @@ def _predict_exact_match(self, entity):
 
         values = []
         for cname in cnames:
-            for item in self._exact_match_mapping["items"][cname]:
+            for item in self._entity_mapping["items"][cname]:
                 item_value = copy.copy(item)
                 item_value.pop("whitelist", None)
                 values.append(item_value)
 
         return values
 
-    def load(self):
-        """Loads the trained entity resolution model from disk."""
-        try:
-            if self._use_text_rel:
-                scoped_index_name = get_scoped_index_name(
-                    self._app_namespace, self._es_index_name
+    def _load(self):
+        self.fit()
+
+
+class SentenceBertCosSimEntityResolver(EntityResolverModelBase):
+    """
+    Resolver class for bert models as described here:
+    https://github.com/UKPLab/sentence-transformers
+    """
+
+    TRANSFORMER_MODEL, POOLING_MODEL, SBERT_MODEL = None, None, None
+
+    def __init__(self, app_path, resource_loader, entity_type, er_config, **_kwargs):
+        super().__init__(app_path, resource_loader, entity_type, er_config)
+        self._entity_mapping = None
+        self._synonyms = None
+        self._synonyms_embs = None
+        self._use_sbert_model = False
+        self._model_pretrained_name_or_abspath = (
+            self.er_config["model_settings"]["pretrained_name_or_abspath"]
+        )
+        self.quantize_model = self.er_config["model_settings"]["quantize_model"]
+
+        # init BERT model
+        if not SentenceBertCosSimEntityResolver.TRANSFORMER_MODEL:
+
+            # load model and quantize if required
+            output_type = self.er_config["model_settings"]["bert_output_type"]
```

Comment:
Maybe do this here:
```
transformer_model, pooling_model, sbert_model = None, None, None
```

then after the try catch, we can either guarantee that all the three above vars are not None, since in the alternate case, the exceptions are raised. If we can do this, then we dont have to do these `if transformer_model else None` type statements:
```
                transformer_model = _get_module_or_attr("torch.quantization", "quantize_dynamic")(
                    transformer_model, {torch_nn_linear}, dtype=torch_qint8
                ) if transformer_model else None
                pooling_model = _get_module_or_attr("torch.quantization", "quantize_dynamic")(
                    pooling_model, {torch_nn_linear}, dtype=torch_qint8
                ) if pooling_model else None
                sbert_model = _get_module_or_attr("torch.quantization", "quantize_dynamic")(
                    sbert_model, {torch_nn_linear}, dtype=torch_qint8
                ) if sbert_model else None
```

Reply:
collapsed all three into a dict and implemented similar to the above style